cmd/run/cmd.go

```diff
@@ -0,0 +1,357 @@
+	}
+
+	if len(args) > 0 {
+		envMap["INITIAL_CLUSTER_LOGIN"] = args[0]
+	}
+
+	var sshAuthSock string
+	if goos == "darwin" {
+		sshAuthSock = "/tmp/ssh/Listeners"
+	} else {
+		sshAuthSock = "/tmp/ssh.sock"
+	}
+	envMap["SSH_AUTH_SOCK"] = sshAuthSock
+	return envMap
+}
+
+func googleCliConfigMounts(homeDir string) []specs.Mount {
+	return []specs.Mount{
+		{
+			Source:      homeDir + "/.config/gcloud/active_config",
+			Destination: "/root/.config/gcloud/active_config_readonly",
+			Options:     []string{"ro"},
+			Type:        define.TypeBind,
+		}, {
+			Source:      homeDir + "/.config/gcloud/configurations/config_default",
+			Destination: "/root/.config/gcloud/configurations/config_default_readonly",
+			Options:     []string{"ro"},
+			Type:        define.TypeBind,
+		}, {
+			Source:      homeDir + "/.config/gcloud/credentials.db",
+			Destination: "/root/.config/gcloud/credentials_readonly.db",
+			Options:     []string{"ro"},
+			Type:        define.TypeBind,
+		}, {
+			Source:      homeDir + "/.config/gcloud/access_tokens.db",
+			Destination: "/root/.config/gcloud/access_tokens_readonly.db",
+			Options:     []string{"ro"},
+			Type:        define.TypeBind,
+		},
+	}
+}
+
+func awsCredentialsMounts(homeDir string) []specs.Mount {
+	return []specs.Mount{
+		{
+			Source:      homeDir + "/.aws/credentials",
+			Destination: "/root/.aws/credentials",
+			Options:     []string{"ro"},
+			Type:        define.TypeBind,
+		},
+		{
+			Source:      homeDir + "/.aws/config",
+			Destination: "/root/.aws/config",
+			Options:     []string{"ro"},
+			Type:        define.TypeBind,
+		},
+	}
+}
+
+func macAgentLocation(fs fileSystemRead, privateTempDir string) (string, error) {
+	dirs, err := fs.ReadDir(privateTempDir)
+	if err != nil {
+		return "", err
+	}
+
+	if len(dirs) < 1 {
+		return "", errors.New(fmt.Sprintf("no dirs found at %v", privateTempDir))
+	}
+
+	for _, dir := range dirs {
+		if dirName := dir.Name(); strings.Contains(dirName, "com.apple.launchd") {
+			return dirName, nil
+		}
+	}
+
+	return "", errors.New(fmt.Sprintf("no dir found at %v containing com.apple.launchd", privateTempDir))
+}
+
+func copyPortmap(fs fileSystemWrite, conn context.Context, containerId string) {
+	tmpdir, err := fs.MkdirTemp("", "occ_portmaps")
+	if err != nil {
+		log.Fatal("Failed to create a tempdir for portmap")
+	}
+	defer fs.RemoveAll(tmpdir)
+
+	data, err := containers.Inspect(conn, containerId, nil)
+	if err != nil {
+		log.Fatal("Failed to inspect container")
+	}
+
+	var hostPorts []string
+	for _, host := range data.NetworkSettings.Ports["9999/tcp"] {
+		hostPorts = append(hostPorts, host.HostPort)
+	}
+
+	portmapFile, err := fs.Create(tmpdir + "/portmap")
+	if err != nil {
+		log.Fatal("Failed to create portmap file")
+	}
+
+	for _, port := range hostPorts {
+		_, err := fmt.Fprintln(portmapFile, port)
+		if err != nil {
+			log.Fatal("Failed to write host port to portmap file")
+		}
+	}
+
+	reader, writer := io.Pipe()
```

Comment:
I really hate this whole section, but I copied it from the podman CLI and it works so I'm running with it.  I've tried simply opening the file with `os.Open` and using the reader returned from that to pass to `containers.CopyFromArchive` below, but I kept getting a 500 error from the podman API.

If someone wants to help me take a look at this I'm more than happy to try and simplify this.